repos/system_upgrade/el7toel8/actors/checkfirewalld/actor.py

```diff
@@ -0,0 +1,37 @@
+from leapp.actors import Actor
+from leapp.models import FirewalldFacts
+from leapp.reporting import Report
+from leapp.libraries.common.reporting import report_with_remediation
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+from leapp.libraries.actor import private
+
+
+class CheckFirewalld(Actor):
+    """
+    Checks for certain firewalld configuration that may prevent an upgrade.
+    """
+
+    name = 'check_firewalld'
+    consumes = (FirewalldFacts,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for facts in self.consume(FirewalldFacts):
+            for table in facts.ebtablesTablesInUse:
+                if not private.isEbtablesTableSupported(table):
+                    report_with_remediation(
+                        title='Firewalld is using an unsupported ebtables table.',
+                        summary='ebtables in RHEL-8 does not support the {} table.'.format(table),
+                        remediation='Remove firewalld direct rules that use ebtables {} table.'.format(table),
+                        severity='high',
+                        flags=['inhibitor'])
+            for ipset_type in facts.ipsetTypesInUse:
+                if not private.isIpsetTypeSupportedByNftables(ipset_type):
+                    report_with_remediation(
+                        title='Firewalld is using an unsupported ipset type.',
+                        summary='ipset type \'{}\' is not supported by firewalld\'s nftables backend.'.format(ipset_type),
+                        remediation='Remove ipsets of type {} from firewalld.'.format(ipset_type),
+                        severity='high',
+                        flags=['inhibitor'])
```

Comment:
Would be possible to generate those messages maximally once? I mean, now you will generate N reports with the same type of issue (one ebtable, another ebtable, ...). We found this problematic in the past from the point:
- generation of duplicit texts
- -> it's less "comprehensible"

The good clue that such messages should be generated just once is the title - when it is same.

Reply:
That's up to you. Each instance will have a different summary/remediation. i.e. the first for loop creates a report for each offending ebtables table instead of the first offender hit. As a user, I would expect a full report. Otherwise a user my fix all the reported issues, then attempt the upgrade process again and be surprised to hit yet new reported issue.